src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -83,62 +46,286 @@ def validate_vm_nics(namespace):
+                           'Microsoft.Network', 'publicIPAddresses'):
+            namespace.public_ip_type = 'existing'
+        else:
+            namespace.public_ip_type = 'new'
+    elif namespace.public_ip_address == '':
+        namespace.public_ip_type = None
+    elif namespace.public_ip_address is None:
+        namespace.public_ip_type = 'new'
+
+def _validate_vm_create_nics(namespace):
+    from azure.cli.core.commands.client_factory import get_subscription_id
+    nics_value = namespace.nics
+    nics = []
 
+    if not nics_value:
+        namespace.nic_type = 'new'
+        return
 
-def validate_default_os_disk(namespace):
-    if not namespace.os_disk_name:
-        namespace.os_disk_name = 'osdisk{}'.format(generate_guid())
+    if not isinstance(nics_value, list):
+        nics_value = [nics_value]
 
+    for n in nics_value:
+        nics.append({
+            'id': n if '/' in n else resource_id(name=n,
+                                                 resource_group=namespace.resource_group_name,
+                                                 namespace='Microsoft.Network',
+                                                 type='networkInterfaces',
+                                                 subscription=get_subscription_id()),
+            'properties': {
+                'primary': nics_value[0] == n
+            }
+        })
 
-def validate_storage_suffix(namespace):
-    from azure.cli.core._profile import CLOUD
-    namespace.storage_suffix = CLOUD.suffixes.storage_endpoint
+    namespace.nics = nics
+    namespace.nic_type = 'existing'
+    namespace.public_ip_type = None
+
+def _validate_vm_create_auth(namespace):
+
+    if not namespace.os_type:
+        raise CLIError("Unable to resolve OS type. Specify '--os-type' argument.")
+
+    if not namespace.authentication_type:
+        # apply default auth type (password for Windows, ssh for Linux) by examining the OS type
+        namespace.authentication_type = 'password' if namespace.os_type == 'windows' else 'ssh'
+
+    if namespace.os_type == 'windows' and namespace.authentication_type == 'ssh':
+        raise CLIError('SSH not supported for Windows VMs.')
+
+    # validate proper arguments supplied based on the authentication type
+    if namespace.authentication_type == 'password':
+        if namespace.ssh_key_value or namespace.ssh_dest_key_path:
+            raise ValueError(
+                "incorrect usage for authentication-type 'password': "
+                "[--admin-username USERNAME] --admin-password PASSWORD")
+
+        if not namespace.admin_password:
+            # prompt for admin password if not supplied
+            import getpass
+            while True:
+                password_1 = getpass.getpass('Admin Password: ')
+                password_2 = getpass.getpass('Confirm Password: ')
+                if password_1 == password_2:
+                    namespace.admin_password = password_1
```

Comment:
and we should output a warning here so it doesn't look like hang? If yes, please just resolve that issue